internal/core/src/segcore/SegmentSealed.h

```diff
@@ -53,14 +57,38 @@ class SegmentSealed : public SegmentInternalInterface {
 
     virtual index::IndexBase*
     GetJsonIndex(FieldId field_id, std::string path) const override {
-        JSONIndexKey key;
-        key.field_id = field_id;
-        key.nested_path = path;
-        auto index = json_indexings_.find(key);
-        if (index == json_indexings_.end()) {
-            return nullptr;
+        int path_len_diff = std::numeric_limits<int>::max();
+        index::IndexBase* best_match = nullptr;
+        std::string_view path_view = path;
+        for (const auto& index : json_indices) {
+            if (index.field_id != field_id) {
+                continue;
+            }
+            switch (index.cast_type.data_type()) {
+                case JsonCastType::DataType::JSON:
+                    if (path_view.length() < index.nested_path.length()) {
+                        continue;
+                    }
+                    if (path_view.substr(0, index.nested_path.length()) ==
```

Comment:
change to `if path_view.starts_with(index.nested_path)`, and remove the above `if (path_view.length() < index.nested_path.length())`.

Reply:
> change to `if path_view.starts_with(index.nested_path)`, and remove the above `if (path_view.length() < index.nested_path.length())`.

starts_with is supported in c++20 :(